plugins/source/gcp/client/client.go

```diff
@@ -105,14 +111,27 @@ func New(ctx context.Context, logger zerolog.Logger, s specs.Source) (schema.Cli
 
 		c.logger.Info().Msg("listing folder projects..")
 		folderProjects, err := listProjectsInFolders(ctx, c.Services.ResourcemanagerProjectsClient, folderIds)
-		projects = append(projects, folderProjects...)
+		projects = setUnion(projects, folderProjects)
 		if err != nil {
 			return nil, fmt.Errorf("failed to list projects: %w", err)
 		}
+
+	case len(gcpSpec.ProjectFilter) > 0:
+		c.logger.Info().Msg("Listing projects with filter..")
+		projectsWithFilter, err := getProjectsV1WithFilter(ctx, gcpSpec.ProjectFilter, options...)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get projects with filter: %s", err)
```

Comment:
nit: I would use `%w` for error
```suggestion
			return nil, fmt.Errorf("failed to get projects with filter: %w", err)
```